web/src/stream_list.js

```diff
@@ -583,9 +582,10 @@ export function update_dom_with_unread_counts(counts) {
             !sub_store.get(stream_id).is_muted &&
             stream_has_any_unread_mention_messages &&
             !stream_has_any_unmuted_unread_mention;
+        const count_info = unread.unread_count_info_for_stream(stream_id);
```

Comment:
Why do you need to do this? I'm pretty sure the `count` object there already has the data we need (`counts` is a `FullUnreadCountsData`, and this is doing duplicate work in a potentially performance-sensitive background update code path.

OK dug into this, and I believe the pre-patch code here was correct, and the explanation for the mystery for how that can possible work given the declared types appears to be is that the return type of `get_counts` was not migrated to TypeScript correctly, because we ignored this comment:
```
                // Note: The format of res.stream_count is completely                                   
                // different from the else clause; with objects                                         
                // containing data for individual topics rather than                                    
                // just muted/unmuted totals, and all muted                                             
                // streams/topics included so that the inbox view can                                   
                // easily show/hide topics when its filters are                                         
                // adjusted.  This should be refactored before the                                      
                // TypeScript migration for this file.                                                  
                stream_counts_by_id.set(stream_id, topic_unread);                                       
                stream_unread_messages += stream_count;                                                 
            } else {                                                                                    
                // get_stream_count_info calculates both the number of                                  
                // unmuted unread as well as the number of muted                                        
                // unreads.                                                                             
                stream_counts_by_id.set(stream_id, this.get_stream_count_info(stream_id));              
                stream_unread_messages += stream_counts_by_id.get(stream_id).unmuted_count;             
                followed_topic_unread_messages += stream_counts_by_id.get(stream_id).followed_count;    
```

Reply:
Oh thanks for catching that!! I totally missed that earlier, sorry. I didn't realize that Typescript couldn't catch something like that -- it just totally accepted two different types of maps without warning.

Here's a PR separating those functionalities #29033